vocabularies/Common.json

```diff
@@ -1288,6 +1288,132 @@
       "@Common.Experimental": true,
       "@Core.IsURL": true,
       "@Core.Description": "Base URL for WebSocket connections"
+    },
+    "RecommendationsFunction": {
+      "$Kind": "Term",
+      "$Type": "Common.QualifiedName",
+      "$AppliesTo": ["EntityType"],
+      "@Common.Experimental": true,
+      "@Core.Description": "Qualified name of a function that computes AI-based recommendations for entities of the annotated type (see [template](#Template_RecommendationsFunction) and [template for asynchronous OData requests](#Template_AsyncRecommendationsFunction))"
+    },
+    "Template_RecommendationsFunction": [
+      {
+        "$Kind": "Function",
+        "$IsBound": true,
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for functions that compute AI-based recommendations for the bound entity and its related entities",
+        "@Core.LongDescription": "If the function is invoked repeatedly in a stateful session, it SHOULD NOT\nrecompute recommendations whose predictors have not changed since the previous invocation.\n\nThe template function itself cannot be invoked.",
+        "$Parameter": [
+          { "$Name": "Entity", "$Type": "Edm.EntityType" },
+          {
+            "$Name": "Targets",
+            "$Collection": true,
+            "@Core.IsURL": true,
+            "@Core.Description": "URLs (relative to the request URL) that address entities, properties or action/function parameters for which recommendations shall be retrieved",
+            "@Core.LongDescription": "The request `SalesOrder('A')/ns.RecommendationsFunction?Targets=[\"Items(10)\"]`\n            retrieves recommendations for the entity `SalesOrder('A')/Items(10)`."
+          }
+        ],
+        "$ReturnType": {
+          "@Core.IsURL": true,
+          "@Core.Description": "URL (relative to the service URL) for retrieving the recommendations (which are computed asynchronously)",
+          "@Core.Example": {
+            "@odata.type": "https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml#Core.PrimitiveExampleValue",
+            "Description": "Relative URL for invoking a function import of the [retrieval function](#Template_RecommendationsRetrievalFunction)",
+            "Value": "RetrieveRecommendations?ID=af43c2"
```

Comment:
@uhlmannm - I believe we will have split this and then make a call via context.execute() or we can directly make a call to this URL via the model?

Reply:
As this is a function, you could directly bind the URL. It would not be an operation binding then and `execute` could not be called. It would rather behave like a normal context binding, i.e. `requestObject` should do the trick. Refreshing via `refresh` would also be possible.
You could also split the Url and use an operation binding. But that should be unnecessary effort.